### PR TITLE
chore(cli-integ): remove in-source compilation of 'bin' directory

### PR DESCRIPTION
**Pull request by rix0rrr**
The CLI integ test package relies on the `bin` directory being compiled in-source. In new projen-managed setups, this is not common anymore.

Move the actual CLI sources into the `lib/cli` directory and launch them from there.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/cx-api/FEATURE_FLAGS.md [hunk 1]
```diff
@@ -93,71 +93,7 @@ The following json shows the current recommended set of flags, as `cdk init` wou
 
 <!-- BEGIN json -->
 ```json
-{
```
Comment (iliapolo) [inline packages/aws-cdk-lib/cx-api/FEATURE_FLAGS.md:96]: Doesn't seem related to the PR? is this intentional?

Comment (rix0rrr) [reply packages/aws-cdk-lib/cx-api/FEATURE_FLAGS.md:96]: Nope

**Review by iliapolo — APPROVED** [review]
